decorator.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9]
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** In keeping with the Ruby convention, each class should be in its own file. I would recommend you move the **CapitalizeDecorator** and **TrimmerDecorator** classes to their own respective files

Reply:
thanks @TedLivist  , I was trying to do this trial  ...person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name ...


I would appreciate advice on which file I should call it from

Reply:
Hi @MICHAELMUNAVU83 You can do it in any file and just import the needed classes.

But I would recommend you can create something of a test file and import any needed classes there, and run your trial. This way you don't need to touch the original classes (keeping it clean). And if you don't need the test file anymore you can just delete it.